lib/bolt/plugin/task.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Bolt
+  class Plugin
+    class Task
+      def hooks
+        %w[inventory_targets inventory_config]
+      end
+
+      def name
+        'task'
+      end
+
+      # This creates it's own PAL so we don't have to pass a promise around
+      #
+      def initialize(config)
+        @config = config
+      end
+
+      attr_reader :config
+
+      def pal
+        # Hiera config should not be used yet.
+        @pal ||= Bolt::PAL.new(config.modulepath, config.hiera_config)
+      end
+
+      def executor
+        # Analytics should be handled at a higher level so create a new executor.
+        @executor ||= Bolt::Executor.new
+      end
+
+      def inventory
+        @inventory ||= Bolt::Inventory.new({}, config)
+      end
+
+      def run_task(opts)
+        result = pal.run_task(opts['task'],
+                              'localhost',
+                              opts['parameters'] || {},
+                              executor,
+                              inventory).first
+
+        # TODO: what kind of error is this?
+        raise result.error_hash['msg'] if result.error_hash
+        result
+      end
+
+      def validate_options(opts); end
+
+      def inventory_config(opts)
+        result = run_task(opts)
+
+        unless result.value.include?('config')
+          raise Bolt::Inventory::ValidatonError, "Task result did not return 'config': #{result.value}"
```

Comment:
```suggestion
          raise Bolt::Inventory::ValidationError, "Task result did not return 'config': #{result.value}"
```

Reply:
That error class also has a second argument for `offending_group` which should be optional.

Reply:
I think making the error message clearer is probably adequate since we print the result as part of this.